ext/nnstreamer/tensor_filter/tensor_filter_llamacpp.cc

```diff
@@ -248,27 +250,21 @@ TensorFilterLlamaCpp::invoke (const GstTensorMemory *input, GstTensorMemory *out
  * The memory allocated for the output tensor data using strndup
  * must be freed by the caller of this function to avoid memory leaks.
  */
-GstTensorMemory *
-TensorFilterLlamaCpp::createOutputTensor (
-    const char *buf, int n, GstTensorFilterProperties *prop)
+bool
+TensorFilterLlamaCpp::createOutputTensor (const std::string &buf,
+    GstTensorMemory *output, GstTensorFilterProperties *prop)
 {
-  if (buf == nullptr || n <= 0 || prop == nullptr) {
+  if (buf.empty () || output == nullptr || prop == nullptr) {
     ml_loge ("Invalid arguments passed to createOutputTensor");
-    return nullptr;
+    return false;
   }
 
-  GstTensorMemory *output = g_new0 (GstTensorMemory, 1);
-  if (!output) {
-    ml_loge ("createOutputTensor: Allocation failed");
-    return nullptr;
-  }
+  output[0].size = buf.length (); /* Stores string length, excluding NULL appended to the end of the string */
+  output[0].data = strndup (buf.c_str (), buf.length ());
 
-  output[0].data = g_strndup (buf, n);
-  output[0].size = n; /* Stores string length, excluding NULL appended to the end of the string */
   if (output[0].data == nullptr) {
     ml_loge ("createOutputTensor: strndup failed");
-    g_free (output);
-    return nullptr;
+    return false;
   }
 
   gst_tensors_info_free (&prop->output_meta);
```

Comment:
Do you really need to free meta (gst_tensors_info) every frame? Do you really need to erase tensor name? Reconfiguring dimension does not require free-reinit.

Reply:
Thank you. I've updated.